components/scream/src/physics/shoc/shoc_functions_f90.cpp

```diff
@@ -2417,6 +2427,57 @@ void pblintd_height_f(Int shcol, Int nlev, Real* z, Real* u, Real* v, Real* usta
   // TODO
 }
 
+void pblintd_init_f(Int shcol, Int nlev, Real* z, bool* check, Real* rino, Real* pblh)
+{
+  using SHOC       = Functions<Real, DefaultDevice>;
+  using Spack      = typename SHOC::Spack;
+  using Scalar     = typename SHOC::Scalar;
+  using Pack1      = typename ekat::Pack<Real, 1>;
+  using view_1d    = typename SHOC::view_1d<Pack1>;
+  using view_2d    = typename SHOC::view_2d<Spack>;
+
+  Kokkos::Array<view_2d, 2> views_2d;
+  ekat::host_to_device({z, rino}, shcol, nlev, views_2d, true);
+
+  view_2d z_2d   (views_2d[0]),
+          rino_2d(views_2d[1]);
+
+  view_1d pblh_d("rino", shcol);
+
+  SHOC::view_1d<bool> check_d("check", shcol);
+  const auto host_check_d = Kokkos::create_mirror_view(check_d);
+
+  Kokkos::parallel_for("pblintd_init", shcol, KOKKOS_LAMBDA (const int& i) {
+
+    Scalar pblh_s{0.};
+    bool check_s{false};
+
+    const auto z_1d     = ekat::subview(z_2d, i);
+    const auto rino_1d  = ekat::subview(rino_2d, i);
+
+    const auto nlev_pack = ekat::npack<Spack>(nlev)-1;
+    const int nlev_indx  = (nlev-1)%Spack::n;
+    Scalar& rino_s        = rino_1d(nlev_pack)[nlev_indx];
+    Scalar& z_s           = z_1d(nlev_pack)[nlev_indx];
+
+    SHOC::pblintd_init(z_s, check_s, rino_s, pblh_s);
+
+    check_d(i)   = check_s;
+    pblh_d(i)[0] = pblh_s;
+  });
+
+  Kokkos::Array<view_2d, 1> out_2d_views = {rino_2d};
+  ekat::device_to_host<int, 1>({rino}, {shcol}, {nlev}, out_2d_views, true);
```

Comment:
You don't need to use the API that takes vectors of sizes if they are all the same.